internal/pkg/gateway/registry.go

```diff
@@ -200,13 +202,42 @@ func (reg *registry) endorsersByOrg(channel string, chaincode string) map[string
 }
 
 func (reg *registry) channelMembers(channel string) gossipdiscovery.Members {
-	return reg.discovery.PeersOfChannel(gossipcommon.ChannelID(channel))
+	members := reg.discovery.PeersOfChannel(gossipcommon.ChannelID(channel))
+
+	// Ensure local endorser ledger height is up-to-date
+	for _, member := range members {
+		if reg.isLocalEndorserID(member.PKIid) {
+			if ledgerHeight, ok := reg.localLedgerHeight(channel); ok {
+				member.Properties.LedgerHeight = ledgerHeight
+			}
+
+			break
+		}
+	}
+
+	return members
 }
 
 func (reg *registry) isLocalEndorserID(pkiID gossipcommon.PKIidType) bool {
 	return !pkiID.IsNotSameFilter(reg.localEndorser.pkiid)
 }
 
+func (reg *registry) localLedgerHeight(channel string) (height uint64, ok bool) {
+	ledger, err := reg.localProvider.Ledger(channel)
+	if err != nil {
+		reg.logger.Debugw("local endorser is not a member of channel", "channel", channel, "err", err)
+		return 0, false
+	}
+
+	info, err := ledger.GetBlockchainInfo()
+	if err != nil {
+		logger.Debugw("failed to get local ledger info", "err", err)
```

Comment:
This should be an error, no?

Reply:
Pushed an update with the changes to log level